codegen/module_system.go

```diff
@@ -544,6 +573,57 @@ func (g *httpClientGenerator) Generate(
 	}, nil
 }
 
+// PopulateQPSLevels loops through endpoint dir and gets qps levels
+func PopulateQPSLevels(EndpointsBaseDir string) map[string]int {
+	qpsLevels := make(map[string]int)
+	endpointFiles := GetListOfAllFilesInEndpointDir(EndpointsBaseDir)
+	for _, endpointFile := range endpointFiles {
+		var config map[string]interface{}
+		bytes, _ := ioutil.ReadFile(endpointFile)
+		err := yaml.Unmarshal(bytes, &config)
+		if err != nil {
```

Comment:
We can unmarshal here by checking the file type:

fileExtension := filepath.Ext(path)

if fileExtension == ".json" {
    // Read JSON
}